### PR TITLE
Add JobVersion To Both Pipelines

### PR DESCRIPTION
**Pull request by QuanWanxx**

### CODE REVIEW

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomJobVersionManager.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.Health.Fhir.Synapse.Core.Jobs.Models;
+
+namespace Microsoft.Health.Fhir.Synapse.Core.Jobs
+{
+    public class DicomJobVersionManager
```
Comment (hongyh13) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomJobVersionManager.cs:11]: same here, rename to `DicomToDataLakeJobVersionManager`

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomJobVersionManager.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            JobVersion.V1,
+        };
+
+        // The job version is added in input data to handle possible version compatibility issues when the version is updated. It is not related to the job definition, so we need to remove the job version property when calculate job identifier.
+        // job version V1
+        public static readonly List<string> DicomToDataLakeOrchestratorJobIdentifierPropertiesV1 = new ()
+        {
+            nameof(DicomToDataLakeOrchestratorJobInputData.JobType),
+            nameof(DicomToDataLakeOrchestratorJobInputData.TriggerSequenceId),
+            nameof(DicomToDataLakeOrchestratorJobInputData.StartOffset),
```
Comment (hongyh13) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomJobVersionManager.cs:28]: what is the consideration to exclude `EndOffset` in orchestrator job while include it in processing job?

**Review by hongyh13 — COMMENTED** [review]
we also need to set the `jobversion` for orchestrator input data and current trigger entity in dicom scheduler service

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomJobVersionManager.cs (hunk 1 above)
Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomJobVersionManager.cs:11]: Updated, thanks.

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomJobVersionManager.cs (hunk 2 above)
Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomJobVersionManager.cs:28]: Both be removed, thanks.

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirJobVersionManager.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.Health.Fhir.Synapse.Core.Jobs.Models;
+
+namespace Microsoft.Health.Fhir.Synapse.Core.Jobs
+{
+    public static class FhirJobVersionManager
```
[the remark this answers is not in the document]
Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirJobVersionManager.cs:11]: Sure, updated.

**Comment by QuanWanxx** [comment]
> we also need to set the `jobversion` for orchestrator input data and current trigger entity in dicom scheduler service

The orchestrator input data also have **JobVersion**. DICOM and FHIR scheduler service both use raw **CurrentTriggerEntity**.

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomToDatalakeSchedulerService.cs [hunk 4]
```diff
@@ -463,6 +463,7 @@ private async Task<CurrentTriggerEntity> CreateInitialTriggerEntity(Cancellation
                 TriggerSequenceId = 0,
                 StartOffset = InitialStartOffset,
                 EndOffset = latestSequence,
+                JobVersion = DicomToDatalakeJobVersionManager.CurrentJobVersion,
```
Comment (hongyh13) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomToDatalakeSchedulerService.cs:466]: we also need to set the job version for orchestrator job input data and updated trigger entity.

**Review by moria97 — APPROVED** [review]

**Pull request merged**